### ISSUE TITLE
i18n: Streamline Singular/Plural Nouns

### ISSUE THREAD

**Issue by Still34** [stashapp/stash#1924]
TL;DR; the localization table is a mess right now with singulars and plurals being thrown all over the place - maybe it's time to do some cleanup.

##  Long Form
Back when i18n was first implemented, a lot of the terminologies were put under the root key of the localization table. As the project has grown, more and more string categories were created. Now we have a *somewhat* clean section for translators to figure out what the context of the string may be based on the key ID (e.g., `actions.add` -> verb form of `add` might be used, as opposed to noun). 

The problem now is that some of these "legacy" strings are now littering the root section of the localization table, with most of them being nouns that cannot be appropriately placed due to the generic nature of the terms. There are now some implementations use `countables.<entity_type>` or just `<entity_type>/<entity_type>s`. 

We need to figure out a way to consolidate all of these into a single group, where the translator could easily define what a singular entity is or a plural is instead of having to repeat themselves across different strings.

**Comment by 7dJx1qP** [comment]
Can we just move all the root-level singular/plural pairs and put them under `countables` and then update the code accordingly?

**Comment by 7dJx1qP** [comment]
I'm looking through the code and I think it would be useful to identify all the patterns where `<entity_type>/<entity_type>s` comes up.

Does searching for occurrences of `intl.formatMessage` or `FormattedMessage` in `.ts` and `.tsx` files give you every lowest level localization call, or are there other functions?

These would be the simplest to update:
```js
<FormattedMessage id="details" />
```
```js
{FormUtils.renderLabel({
  title: intl.formatMessage({ id: "studio" }),
})}
```

Then there's components where a singular/plural entity are passed in which would require refactoring the component and all usages of the component:
```js
<DeleteEntityDialog
  selected={selectedMovies}
  onClose={onClose}
  singularEntity={intl.formatMessage({ id: "movie" })}
  pluralEntity={intl.formatMessage({ id: "movies" })}
  destroyMutation={useMoviesDestroy}
/>
```

The most complex case I've found so far is something like:
```js
createStringCriterionOption("details")
```
where you have to go through multiple levels of abstraction before you get to the actual `intl.formatMessage` call

**Comment by 7dJx1qP** [comment]
I was looking at some localization best practices and found another problem with our translation strings that is described [here](https://github.com/folio-org/stripes/blob/master/doc/i18n.md#avoid-substituting-nouns-and-verbs-for-placeholders)

The problem is we have generic translation strings where we substitute in the entity translations, but that probably doesn't allow for proper translations in some languages such as ones that have gendered translations of those entities.
Example:
`"delete_alert": "The following {count, plural, one {{singularEntity}} other {{pluralEntity}}} will be deleted permanently:",`
The translation of this sentence could change depending on the type of entity.

Unfortunately, the solution to this is to make the translation file more verbose with each possible entity having its own entry. This means more translation strings to translate. I don't think this means the noun singulars and plurals shouldn't be consolidated since there's definitely an inconsistency in having both `countables.<entity_type>` and `<entity_type>/<entity_type>s`, but those consolidated entries shouldn't be used as substitutes in other translation strings, which they currently are in many places.

**Comment by Still34** [comment]
Do you think you would be able to take on such a task? Unfortunately, React isn't something I'm very well-versed at, which also means I'm not familiar with its best practices when it comes to i18n structuring.

**Comment by 7dJx1qP** [comment]
I'm not experienced with React or i18n either, so I'd like some input from the main devs to see if this would be going in the right direction.

Back when I first saw this issue, I started work on a branch making some changes just to see what it would be like, so I could put that up as a draft PR and get some feedback on it.

**Comment by Roy-Wanyoike** [comment]
Wanted to add a concrete data point here as a contributor who just opened a Kiswahili (Swahili, `sw-KE`) translation PR — #7252 — and ran straight into this issue from the translator's side.

## What actually happened during translation

Swahili is a Bantu language with a **noun-class system** (8+ classes) rather than an Indo-European singular/plural distinction. For most of the `countables.*` entries that use the `{count, plural, one {…} other {…}}` form, the `one` and `other` branches collapse to the **same string** in Swahili because the noun doesn't inflect for count the way English "File/Files" does.

Concretely, in `sw-KE.json`:

```json
"countables": {
  "files":  "{count, plural, one {Faili} other {Faili}}",
  "images": "{count, plural, one {Picha} other {Picha}}"
}
```

This is **not** a translation error — "Faili" (loanword from English "file") and "Picha" (image/picture) genuinely don't distinguish singular/plural in everyday Kenyan/Tanzanian usage. The English-style `one`/`other` dichotomy simply doesn't apply to these words, and forcing the translator to pick two different strings would produce something artificial.

This isn't unique to Swahili — it affects every Bantu language (Zulu, Xhosa, Kikuyu, Lingala, Kinyarwanda, etc.), many Austronesian languages (Indonesian already in stash likely has the same issue), Turkish, Japanese, Korean, and Chinese — i.e. roughly half the languages stash already supports.

## A concrete direction worth considering

`react-intl` v6 (which stash already uses) fully supports ICU's `{count, plural, …}` form, and ICU itself is happy with `one`/`other` branches that produce identical output. So the current setup *works* — it just produces slightly odd-looking JSON for non-Indo-European languages.

A low-friction consolidation that would help translators without a big refactor:

1. **Audit & migrate all root-level `<entity>` / `<entity>s` pairs into `countables.<entity>`** using the `{count, plural, one {…} other {…}}` form (as @7dJx1qP suggested above). This gives translators a single, predictable place to define count behaviour per entity.
2. **Keep `one`/`other` as the only two branches** (don't try to model full Bantu noun classes in ICU — that's a rabbit hole). Translators for languages that don't inflect simply repeat the same string in both branches, which is valid and readable.
3. **Document the convention in `ui/v2.5/src/locales/README.md`** so future contributors (like me, last week) don't have to reverse-engineer it from `ar.json` / `zh-CN.json`.

A more ambitious direction would be adopting `{count, plural, zero {…} one {…} two {…} few {…} many {…} other {…}}` for languages that need it (Arabic, Russian, Polish already in stash benefit from `zero`/`few`/`many`), but that's a separate, larger piece of work.

## What I'm NOT proposing

I'm not volunteering to do the full migration in this comment — it's a multi-file refactor that needs maintainer sign-off on the target shape first. But I'm happy to:
- Help scope a concrete migration PR (enumerate every root-level singular/plural pair and propose the `countables.*` target key for each) if a maintainer wants to greenlight the direction.
- Be a test case for the new convention by retrofitting `sw-KE.json` once the consolidation lands.

Concrete translator-side data point in case it's useful for the decision. cc @WithoutPants in case there's a preferred direction already.
